src/main/java/core/basesyntax/service/operation/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.StorageService;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private final StorageService storageService;
+
+    public PurchaseOperationHandler(StorageService storageService) {
+        this.storageService = storageService;
+    }
+
+    @Override
+    public void handle(FruitTransaction transaction, StorageService storageService) {
+        int currentQuantity = storageService.getFruitQuantity(transaction.getFruit());
+        storageService.addFruit(transaction.getFruit(), -transaction.getQuantity());
+    }
```

Comment:
According to the checklist, you should check the result balance before saving it in the Storage - it should be positive. Throw a RuntimeException in case the balance is negative. Currently, the implementation allows for a negative balance, which is incorrect.